crates/test-programs/src/bin/async_http_echo.rs

```diff
@@ -44,18 +43,25 @@ impl Handler for Component {
 
             async_support::spawn(async move {
                 let mut body_rx = body.stream().unwrap();
-                while let Some(Ok(chunk)) = body_rx.next().await {
-                    pipe_tx.send(chunk).await.unwrap();
+                let mut chunk = Vec::with_capacity(1024);
+                loop {
+                    let (status, buf) = body_rx.read(chunk).await;
+                    chunk = buf;
+                    match status {
+                        StreamResult::Complete(_) => {
+                            chunk = pipe_tx.write_all(chunk).await;
+                            assert!(chunk.is_empty());
+                        }
+                        StreamResult::Closed => break,
+                        // TODO: this should panic
+                        StreamResult::Cancelled => {}
```

Comment:
@dicej this is one thing we'll need to discuss, if this branch panics, which it should, then the test fails